packages/typeahead/components/Typeahead.tsx

```diff
@@ -181,7 +181,7 @@ class Typeahead extends React.PureComponent<TypeaheadProps, TypeaheadState> {
                         );
                       },
                       value: value === undefined ? inputValue : value,
-                      downshiftReset: other.reset,
+                      ...(downshiftReset && { downshiftReset: other.reset }),
```

Comment:
will this attempt to spread `false` if downshiftReset is undefined? Maybe you want `...(downshiftReset ? { downshiftReset: other.reset } : {})`?

Not sure, if you tested and it's okay, then 👍🏻

Reply:
This should be fine. The spread operator takes lower precedence than the `&&` operator. If the conditional returns false then the line will be reduced to `...(false)`, which shouldn't cause any issues because the spread operator ignores values without a property. [See this article for more info](https://medium.com/@slamflipstrom/conditional-object-properties-using-spread-in-javascript-714e0a12f496). 

However, I've just realised that this isn't going to work the way I need it to. It's a bit complicated and wordy to type out. I'm going to sleep on it and see if I can find another solution tomorrow.